bionic/cache_api.py

```diff
@@ -0,0 +1,108 @@
+"""
+Contains the Cache class, which provides a user-facing API for directly interacting
+with Bionic's cache.
+"""
+
+from functools import total_ordering
+
+from .descriptors.parsing import entity_dnode_from_descriptor
+from .persistence import path_from_url, is_file_url
+
+
+class Cache:
+    """
+    A programmatic interface to Bionic's persistent cache.
+
+    Accessible as an attribute named ``cache`` on a :class:`Flow <bionic.Flow>`
+    object. Use ``get_entries`` to iterate through the set of cache entries.
+    """
+
+    def __init__(self, flow):
+        self._flow = flow
```

Comment:
Good point. I was originally planning to depend on the flow for other stuff, like checking whether a cache entry is used by the current flow. But for the current implementation we don't need this.

Reply:
Hmm, fixing this will require rearranging the way this is initialized in Flow, because we want to call `deriver.get_ready()` at the last minute rather than in the constructor of `Flow` (because we don't want to do all the deriver setup multiple times when we do `flow.setting(...).setting(...)...`). Since I expect us to use the flow itself in the future, I'll just leave a comment here explaining that.